src/ranged.cpp

```diff
@@ -341,29 +424,52 @@ dealt_projectile_attack Creature::projectile_attack( const projectile &proj_arg,
     return attack;
 }
 
+static double erfinv( double x )
+{
+    const double epsilon = 1e-6;
+    double z = 0.0;
+
+    // find erf(z) - x = 0
+    // d/dz ( erf(z) - x ) = 2/sqrt(pi) . e^(-z^2)
+
+    for( int n = 0; n < 10; ++n ) {
```

Comment:
Yeah, this produces the smoother graph.

There are various ways to compute erfinv but it is a transcendental function so there is no analytic exact way to do it. This is a Newton-Raphson approximation that relies on already having erf. You can also do it with a big higher-order polynomial with lots of magic constants.

For our purposes, if iteration is a problem then the simplest thing here would be to short-circuit the x = 0.5 case and return a constant since that's almost all of the calls.